### PR TITLE
feat(build base): build base

### PR DESCRIPTION
**Pull request by gavin1ee**

### CODE REVIEW

File: src/Services/MASA.Scheduler.Server/Actors/OrderActor.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿namespace MASA.Scheduler.Service.Actors
+{
+    public class JobActor : Actor, IJobActor
```
Comment (MayueCif) [inline src/Services/MASA.Scheduler.Server/Actors/OrderActor.cs:3]: class name diff file name

File: src/Services/MASA.Scheduler.Server/Program.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    .AddTransient(typeof(IMiddleware<>), typeof(LogMiddleware<>))
+    .AddFluentValidation(options =>
+    {
+        options.RegisterValidatorsFromAssemblyContaining<Program>();
+    })
+    .AddTransient(typeof(IMiddleware<>), typeof(ValidatorMiddleware<>))
+    .AddDomainEventBus(options =>
+    {
+        options.UseEventBus()
+               .UseUoW<ShopDbContext>(dbOptions => dbOptions.UseSqlite("DataSource=:memory:"))
```
Comment (MayueCif) [inline src/Services/MASA.Scheduler.Server/Program.cs:58]: to usesqlserver

**Review by MayueCif — APPROVED** [review]

File: MASA.Scheduler.sln [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "dapr", "dapr", "{E011B243-0389-468B-B903-83445BED69A4}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "components", "components", "{FBFB499A-7022-4417-881B-68F63DBDB7F4}"
+	ProjectSection(SolutionItems) = preProject
+		dapr\components\masa-scheduler-pubsub.yaml = dapr\components\masa-scheduler-pubsub.yaml
+		dapr\components\masa-scheduler-statestore.yaml = dapr\components\masa-scheduler-statestore.yaml
+	EndProjectSection
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Masa.Scheduler.Server", "src\Services\Masa.Scheduler.Server\Masa.Scheduler.Server.csproj", "{87A8E21E-DF58-4BEA-8EA1-EF333BE4D2B3}"
```
Comment (doddgu) [inline MASA.Scheduler.sln:18]: Masa.Scheduler.Services.Server

File: MASA.Scheduler.sln [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Web", "Web", "{F3986DD3-0372-4929-B72A-09D93196541E}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ApiGateways", "ApiGateways", "{32774681-0D67-48A6-A379-D57015094F85}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Masa.Scheduler.Caller", "src\ApiGateways\Caller\Masa.Scheduler.Caller\Masa.Scheduler.Caller.csproj", "{40685EC0-64DF-4742-B8F6-602AFAEFC0EA}"
```
Comment (doddgu) [inline MASA.Scheduler.sln:24]: Masa.Scheduler.ApiGateways.Caller

File: MASA.Scheduler.sln [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+		Directory.Build.props = Directory.Build.props
+	EndProjectSection
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Masa.Scheduler.Web.Admin.Server", "src\Web\MASA.Scheduler.Web.Admin.Server\Masa.Scheduler.Web.Admin.Server.csproj", "{F57C6686-881E-45C9-B25D-BBE943F43A1D}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Masa.Scheduler.Web.Admin.WebAssembly", "src\Web\MASA.Scheduler.Web.Admin.WebAssembly\Masa.Scheduler.Web.Admin.WebAssembly.csproj", "{0512CA4E-9130-41B6-8717-40C4E991A005}"
+EndProject
+Project("{E53339B2-1760-4266-BCC7-CA923CBCF16C}") = "docker-compose", "docker-compose.dcproj", "{F17CC0A5-E77B-48A8-907C-90CB0020F869}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Masa.Scheduler.Worker", "src\Services\Masa.Scheduler.Worker\Masa.Scheduler.Worker.csproj", "{B220B87B-3109-4633-B1DC-0C3CAF148C84}"
```
Comment (doddgu) [inline MASA.Scheduler.sln:40]: Masa.Scheduler.Services.Worker

File: src/ApiGateways/Caller/MASA.Scheduler.Caller/Callers/JobCaller.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿using MASA.Scheduler.Contracts.Server.Model;
+
+namespace MASA.Scheduler.Caller.Callers
+{
+    public class JobCaller : HttpClientCallerBase
+    {
+        protected override string BaseAddress { get; set; } = "http://localhost:6239";
```
Comment (doddgu) [inline src/ApiGateways/Caller/MASA.Scheduler.Caller/Callers/JobCaller.cs:7]: Why is the port 6239?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Callers/JobCaller.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+﻿using MASA.Scheduler.Contracts.Server.Model;
+
+namespace MASA.Scheduler.Caller.Callers
+{
+    public class JobCaller : HttpClientCallerBase
+    {
+        protected override string BaseAddress { get; set; } = "http://localhost:16002";
```
Comment (MayueCif) [inline src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Callers/JobCaller.cs:7]: BaseAddress value from configuration，or refer to Masa auth for actual writing.

File: src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Job.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
+﻿namespace MASA.Scheduler.Service.Domain.Aggregates.Jobs
+{
+    public class Job : AggregateRoot<int>
+    {
+        public Job()
+        {
+        }
+
+        public DateTimeOffset CreationTime { get; set; } = DateTimeOffset.Now;
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Job.cs:9]: private or protected set

File: src/Services/Masa.Scheduler.Services.Worker/Program.cs [hunk 9]
```diff
@@ -0,0 +1,6 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Worker/Program.cs:4]: delete

File: src/Services/Masa.Scheduler.Services.Server/Infrastructure/Repositories/JobRepository.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿namespace MASA.Scheduler.Service.Infrastructure.Repositories
+{
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Infrastructure/Repositories/JobRepository.cs:2]: change to file scope namespace (all)

File: src/Services/Masa.Scheduler.Services.Server/Domain/Services/ProductDomainService.cs [hunk 11]
```diff
@@ -0,0 +1,33 @@
+﻿namespace MASA.Scheduler.Service.Domain.Services
+{
+    public class ProductDomainService : DomainService
+    // you can alse implement an interface like below
+    //, ISagaEventHandler<OrderCreatedDomainEvent>
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Services/ProductDomainService.cs:5]: delete comment

**Review by MayueCif — CHANGES_REQUESTED** [review]
modify with conversation and delete unnecessary img

**Comment by MayueCif** [comment]
@doddgu LGTM

**Review by MayueCif — APPROVED** [review]

File: src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs [hunk 12]
```diff
@@ -0,0 +1,12 @@
+﻿namespace MASA.Scheduler.Contracts.Server.Model;
+
+public class Job
+{
+    public DateTime CreationTime { get; set; }
+
+    public int Id { get; set; }
+
+    public string OrderNumber { get; set; } = "";
```
Comment (doddgu) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs:9]: OrderNumber?

File: src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs [hunk 13]
```diff
@@ -0,0 +1,12 @@
+﻿namespace MASA.Scheduler.Contracts.Server.Model;
```
Comment (doddgu) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs:1]: Model? Dto?

File: src/Services/Masa.Scheduler.Services.Server/Actors/IJobActor.cs [hunk 14]
```diff
@@ -0,0 +1,6 @@
+﻿namespace MASA.Scheduler.Service.Actors;
+
+public interface IJobActor : IActor
```
Comment (doddgu) [inline src/Services/Masa.Scheduler.Services.Server/Actors/IJobActor.cs:3]: Why is actor?

File: src/Services/Masa.Scheduler.Services.Server/Application/Jobs/JobCommandHandler.cs [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+    [EventHandler(Order = 1)]
+    public async Task CreateHandleAsync(JobCreateCommand command)
+    {
+        await _domainService.CreateJobAsync();
+        //you work
+        await Task.CompletedTask;
+    }
+}
+
+public class JobStockHandler : CommandHandler<JobCreateCommand>
```
Comment (doddgu) [inline src/Services/Masa.Scheduler.Services.Server/Application/Jobs/JobCommandHandler.cs:21]: Why need a JobStock?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Callers/JobCaller.cs [hunk 16]
```diff
@@ -0,0 +1,21 @@
+﻿namespace MASA.Scheduler.Caller.Callers;
+
+public class JobCaller : HttpClientCallerBase
```
[the remark this answers is not in the document]
Comment (gavin1ee) [reply src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Callers/JobCaller.cs:3]: Already changed in branch feature/team

File: src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs (hunk 12 above)
Comment (gavin1ee) [reply src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs:9]: Already changed in branch feature/domainAggregates

File: src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs (hunk 13 above)
Comment (gavin1ee) [reply src/Contracts/MASA.Scheduler.Contracts.Server/Model/Job.cs:1]: Already changed in branch feature/domainAggregates

File: src/Services/Masa.Scheduler.Services.Server/Actors/IJobActor.cs (hunk 14 above)
Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Actors/IJobActor.cs:3]: will changed when develop Job module

File: src/Services/Masa.Scheduler.Services.Server/Application/Jobs/Commands/JobCreateCommand.cs [hunk 17]
```diff
@@ -0,0 +1,5 @@
+﻿namespace MASA.Scheduler.Service.Application.Jobs.Commands;
+
+public record JobCreateCommand : DomainCommand
```
[the remark this answers is not in the document]
Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Application/Jobs/Commands/JobCreateCommand.cs:3]: will changed when develop Job module

File: src/Services/Masa.Scheduler.Services.Server/Application/Jobs/JobCommandHandler.cs (hunk 15 above)
[the remark this answers is not in the document]
Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Application/Jobs/JobCommandHandler.cs:21]: will changed when develop Job module

Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Application/Jobs/JobCommandHandler.cs:21]: It is the template create. I will changed when develop Job module

File: src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Job.cs [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public Job(string orderNumber, string address)
+    {
+        OrderNumber = orderNumber;
+        Address = address;
+    }
+
+    public DateTimeOffset CreationTime { get; private set; } = DateTimeOffset.Now;
+
+    public string OrderNumber { get; private set; } = default!;
```
[the remark this answers is not in the document]
Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Job.cs:13]: Already changed in branch feature/domainAggregates

File: src/Services/Masa.Scheduler.Services.Server/Domain/Services/JobDomainService.cs [hunk 19]
```diff
@@ -0,0 +1,27 @@
+﻿namespace MASA.Scheduler.Service.Domain.Services;
+
+public class JobDomainService : DomainService
+{
+    private readonly ILogger<Job> _logger;
+    private readonly IJobRepository _jobRepository;
```
[the remark this answers is not in the document]
Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Domain/Services/JobDomainService.cs:6]: It is the template create. I will changed when develop Job module

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**